pootle/apps/pootle_store/forms.py

```diff
@@ -288,40 +287,40 @@ def clean(self):
-            with update_data_after(unit.store):
+            if not self.updated_fields:
+                return
+            self.instance.changed_with = kwargs.pop("changed_with", None)
+            with update_data_after(self.instance.store):
                 current_time = timezone.now()
                 if SubmissionFields.TARGET in (f[0] for f in self.updated_fields):
-                    unit.submitted_by = self.user
-                    unit.submitted_on = current_time
-                    unit.reviewed_by = None
-                    unit.reviewed_on = None
-                suggestion = self.cleaned_data["suggestion"]
-                user = (
-                    suggestion.user
-                    if suggestion
-                    else self.user)
-                unit.save(
-                    submitted_on=current_time,
-                    submitted_by=user,
-                    changed_with=changed_with)
-                translation_project = unit.store.translation_project
-                for field, old_value, new_value in self.updated_fields:
-                    if field == SubmissionFields.TARGET and suggestion:
-                        old_value = str(suggestion.target_f)
-                    sub = Submission(
-                        creation_time=current_time,
-                        translation_project=translation_project,
-                        submitter=self.user,
-                        unit=unit,
-                        field=field,
-                        type=SubmissionTypes.WEB,
-                        old_value=old_value,
-                        new_value=new_value)
-                    sub.save()
-            return unit
+                    self.instance.submitted_by = self.user
+                    self.instance.submitted_on = current_time
+                    self.instance.reviewed_by = None
+                    self.instance.reviewed_on = None
+                suggestion = self.cleaned_data['suggestion']
+                reviewed_by = None
+                if suggestion:
+                    suggestion.unit = self.instance
+                    review.get(Suggestion)(
+                        [suggestion],
+                        self.user,
+                        SubmissionTypes.WEB).accept()
+                    reviewed_by = self.user
+                    if self.cleaned_data['comment']:
+                        ckwargs = dict(
+                            comment=self.cleaned_data['comment'],
+                            user=self.user)
+                        comment_form = UnsecuredCommentForm(suggestion, ckwargs)
+                        if comment_form.is_valid():
+                            comment_form.save()
+                else:
+                    kwargs["commit"] = False
+                    unit = super(UnitForm, self).save(*args, **kwargs)
+                    unit.save(
+                        changed_with=SubmissionTypes.WEB,
+                        reviewed_by=reviewed_by,
```

Comment:
Because it is set as such in https://github.com/translate/pootle/pull/6059/files#diff-0b3fa96668cdb6d4000baa9e4dfa58abR301